src/lib/Types/Core.hs

```diff
@@ -318,7 +318,8 @@ data TopEnv (n::S) = TopEnv
   , envCustomRules :: CustomRules n
   , envCache :: Cache n
   , envLoadedModules :: LoadedModules n
-  , envLoadedObjects :: LoadedObjects n }
+  , envLoadedObjects :: LoadedObjects n
+  , envDynamicVarStores :: DynamicVarStores }
```

Comment:
The fact that this made its was all the way to the `Env` is very surprising and feels like a mistake to me. I thought we're just going to manage that within the topper monad and nothing else in Dex should ever have to think about this. Access to dynamically scoped values is realized through the special primitive ops, so you should never have to carry this here.

Reply:
Yes, I think I agree. I'll move things around so that we store the dynamic vars (and maybe even the other pointers) somewhere separate.